spy/vm/vm.py

```diff
@@ -102,7 +102,7 @@ class SPyVM:
     bluecache: BlueCache
     emit_warning: Callable[[SPyError], None]
     # For use by --colorize to remember the red/blue color of each expr
-    expr_color_map: Optional[dict[ast.Expr, Color]]
+    ast_color_map: Optional[dict[ast.Expr | ast.Assign, Color]]
```

Comment:
I think we can just use `ast.Node` here, as it's the common superclass of both.
Also, I suspect that in the future we will add more `blue` statements (e.g. I plan to introduce `for x in UNROLL(some_blue_iterabale)` which will be unrolled during redshift).